x/evm/keeper/integration_test.go

```diff
@@ -1,246 +1,425 @@
+					factory.ContractDeploymentData{
+						Contract:        compiledContract,
+						ConstructorArgs: constructorArgs,
 					},
-					Entry("legacy tx", func() txParams {
-						return txParams{0, big.NewInt(baseFee), nil, nil, nil}
-					}),
-					Entry("dynamic tx", func() txParams {
-						return txParams{0, nil, big.NewInt(baseFee), big.NewInt(0), &ethtypes.AccessList{}}
-					}),
 				)
+				Expect(err).To(BeNil())
+				Expect(contractAddr).ToNot(Equal(common.Address{}))
+
+				err = s.network.NextBlock()
+				Expect(err).To(BeNil())
 			})
 
-			//nolint:all
-			Context("during DeliverTx", func() {
-				DescribeTable("should accept transactions with gas Limit > 0",
-					func(malleate getprices) {
-						p := malleate()
-						to := utiltx.GenerateAddress()
-						msgEthereumTx := buildEthTx(privKey, &to, p.gasLimit, p.gasPrice, p.gasFeeCap, p.gasTipCap, p.accesses)
-						res, err := testutil.DeliverEthTx(s.app, privKey, msgEthereumTx)
-						Expect(err).To(BeNil())
-						Expect(res.IsOK()).To(Equal(true), "transaction should have succeeded", res.GetLog())
-					},
-					Entry("legacy tx", func() txParams {
-						return txParams{100000, big.NewInt(baseFee), nil, nil, nil}
-					}),
-					Entry("dynamic tx", func() txParams {
-						return txParams{100000, nil, big.NewInt(baseFee), big.NewInt(0), &ethtypes.AccessList{}}
-					}),
-				)
-				DescribeTable("should not accept transactions with gas Limit > 0",
-					func(malleate getprices) {
-						p := malleate()
-						to := utiltx.GenerateAddress()
-						msgEthereumTx := buildEthTx(privKey, &to, p.gasLimit, p.gasPrice, p.gasFeeCap, p.gasTipCap, p.accesses)
-						res, err := testutil.DeliverEthTx(s.app, privKey, msgEthereumTx)
-						Expect(err).ToNot(BeNil(), "transaction should have failed", res.GetLog())
+			DescribeTable("Executes a contract call", func(getTxArgs func() evmtypes.EvmTxArgs) {
+				senderPriv := s.keyring.GetPrivKey(0)
+				compiledContract := contracts.ERC20MinterBurnerDecimalsContract
+				recipientKey := s.keyring.GetKey(1)
+
+				// Execute contract call
+				mintTxArgs := getTxArgs()
+				mintTxArgs.To = &contractAddr
+
+				amountToMint := big.NewInt(1e18)
+				mintArgs := factory.CallArgs{
+					ContractABI: compiledContract.ABI,
+					MethodName:  "mint",
+					Args:        []interface{}{recipientKey.Addr, amountToMint},
+				}
+				mintResponse, err := s.factory.ExecuteContractCall(senderPriv, mintTxArgs, mintArgs)
+				Expect(err).To(BeNil())
+				Expect(mintResponse.IsOK()).To(Equal(true), "transaction should have succeeded", mintResponse.GetLog())
+
+				// Check contract call response has the expected topics for a mint
+				// call within an ERC20 contract
+				expectedTopics := []string{
+					"0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef",
+					"0x0000000000000000000000000000000000000000000000000000000000000000",
+				}
+				err = integrationutils.CheckTxTopics(mintResponse, expectedTopics)
```

Comment:
I don't find this useful. It'd be better to decode the logs and topics and compare them with the real values instead of just comparing a blob of unreadable strings.

Can we do that instead?

Reply:
I agree on this, I need to create an organized way to manage this through the contracts package that we are using. So that every contract has a constant with the respective events every method should emit or something like that. Will add this in another PR